tests/TestCase/Controller/Component/RequestHandlerComponentTest.php

```diff
@@ -899,217 +865,6 @@ public function testPrefers()
     }
 
     /**
-     * test that AJAX requests involving redirects trigger requestAction instead.
-     *
-     * @return void
-     * @triggers Controller.beforeRedirect $this->Controller
-     */
-    public function testAjaxRedirectAsRequestAction()
-    {
-        Configure::write('App.namespace', 'TestApp');
-        Router::connect('/:controller/:action');
-        $event = new Event('Controller.beforeRedirect', $this->Controller);
-
-        $this->Controller->RequestHandler = new RequestHandlerComponent($this->Controller->components());
-        $this->Controller->request = $this->getMockBuilder('Cake\Network\Request')
-            ->setMethods(['is'])
-            ->getMock();
-        $this->Controller->response = $this->getMockBuilder('Cake\Network\Response')
-            ->setMethods(['_sendHeader', 'stop'])
-            ->getMock();
-        $this->Controller->RequestHandler->request = $this->Controller->request;
-        $this->Controller->RequestHandler->response = $this->Controller->response;
-        $this->Controller->request->expects($this->any())->method('is')->will($this->returnValue(true));
-
-        $response = $this->Controller->RequestHandler->beforeRedirect(
-            $event,
-            ['controller' => 'RequestHandlerTest', 'action' => 'destination'],
-            $this->Controller->response
-        );
-        $this->assertRegExp('/posts index/', $response->body(), 'RequestAction redirect failed.');
-    }
-
-    /**
-     * Test that AJAX requests involving redirects handle querystrings
-     *
-     * @return void
-     * @triggers Controller.beforeRedirect $this->Controller
-     */
-    public function testAjaxRedirectAsRequestActionWithQueryString()
-    {
-        Configure::write('App.namespace', 'TestApp');
-        Router::connect('/:controller/:action');
-        $event = new Event('Controller.beforeRedirect', $this->Controller);
-
-        $this->Controller->RequestHandler = new RequestHandlerComponent($this->Controller->components());
-        $this->Controller->request = $this->getMockBuilder('Cake\Network\Request')
-            ->setMethods(['is'])
-            ->getMock();
-        $this->Controller->response = $this->getMockBuilder('Cake\Network\Response')
-            ->setMethods(['_sendHeader', 'stop'])
-            ->getMock();
-        $this->Controller->RequestHandler->request = $this->Controller->request;
-        $this->Controller->RequestHandler->response = $this->Controller->response;
-        $this->Controller->request->expects($this->any())
-            ->method('is')
-            ->with('ajax')
-            ->will($this->returnValue(true));
-
-        $response = $this->Controller->RequestHandler->beforeRedirect(
-            $event,
-            '/request_action/params_pass?a=b&x=y?ish',
-            $this->Controller->response
-        );
-        $data = json_decode($response, true);
-        $this->assertEquals('/request_action/params_pass', $data['here']);
```

Comment:
Looks like the tests are failing at this point. In the next request the controller name is unknown. 

The only difference I see is that the Request class does not receive $params['Params']['controller']; (and action). That is the point where it fails.